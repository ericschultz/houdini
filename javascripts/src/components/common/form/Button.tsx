// License: LGPL-3.0-or-later
import * as React from 'react';
import { observer, propTypes } from 'mobx-react';
import {InjectedIntlProps, injectIntl} from 'react-intl';

export interface ButtonProps extends React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>
{
  //should be required but we're on typescript 2
  buttonSize?: 'tiny'|'default'|'large'|'jumbo'
}

class Button extends React.Component<ButtonProps, {}> {

  public static defaultProps = {
    buttonSize: "default"
  };

  winnowProps() : ButtonProps {
    let ourProps = {...this.props}
    delete ourProps.buttonSize
    return ourProps

  }

  render() {
    let className = 'button'
    if (this.props.buttonSize != 'default') {
      className += "--" + this.props.buttonSize
    }
    
    if(this.props.className) {
      className += " " + this.props.className
    }
    return <button {...this.winnowProps()} className={className}/>;
  }
}

export default Button



