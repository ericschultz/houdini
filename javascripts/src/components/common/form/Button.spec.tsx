// License: LGPL-3.0-or-later
import * as React from 'react';
import 'jest';
import Button from './Button'
import  toJson  from 'enzyme-to-json';
import { mount } from 'enzyme';

describe('Button', () => {
  test('tiny', () => {
    expect(toJson(mount(<Button buttonSize="tiny"/>))).toMatchSnapshot()
  })
  
  test('default', () => {
    expect(toJson(mount(<Button buttonSize="default"/>))).toMatchSnapshot()
  })

  test('large', () => {
    expect(toJson(mount(<Button buttonSize="large" className="anotherName Name"/>))).toMatchSnapshot()
  })

  test('jumbo', () => {
    expect(toJson(mount(<Button buttonSize="jumbo" className="anotherName Name">value</Button>))).toMatchSnapshot()
  })
})