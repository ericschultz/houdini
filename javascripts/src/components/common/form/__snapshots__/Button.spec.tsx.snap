// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 1`] = `
<Button
  buttonSize="default"
>
  <button
    className="button"
  />
</Button>
`;

exports[`Button jumbo 1`] = `
<Button
  buttonSize="jumbo"
  className="anotherName Name"
>
  <button
    className="button--jumbo anotherName Name"
  >
    value
  </button>
</Button>
`;

exports[`Button large 1`] = `
<Button
  buttonSize="large"
  className="anotherName Name"
>
  <button
    className="button--large anotherName Name"
  />
</Button>
`;

exports[`Button tiny 1`] = `
<Button
  buttonSize="tiny"
>
  <button
    className="button--tiny"
  />
</Button>
`;
